.github/workflows/weekly-summary.yml

```diff
@@ -0,0 +1,23 @@
+# Generate a weekly summary of test failures
+name: Weekly Summary
+on:
+  schedule:
+    # Run early Monday mornings, but long enough after daily test run for results
+    - cron: 0 8 * * 1
+  # be able to start this action manually from a actions tab when needed
+  workflow_dispatch:
+
+jobs:
+  weekly_summary:
+    name: Create weekly summary of test runs
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v3
+      - name: Install pycurl
+        run: apt-get install -y python3-pycurl
```

Comment:
Needs to be run as root (I hit it when tried to run the workflow locally).
`run: sudo apt-get install -y python3-pycurl`